test_fms/test-lib.sh.in

```diff
@@ -33,6 +33,11 @@ TEST_NAME="$(basename "$0" .sh)"
 TEST_NUMBER="${TEST_NAME%%-*}"
 TEST_NUMBER="${TEST_NUMBER#t}"
 
+# if using intel with a limited stack size, sets to run smaller tests
+if [ "$($FC --version | grep ifort)" -a "$(ulimit -s)" != "unlimited" 2> /dev/null ]; then
```

Comment:
I think this will work for now, but you might need to add some actual numbers in here.  I can envision a small cloud cluster that has stacksize set to unlimited but still can't handle these large arrays.  I can also envision a system with a large enough default stacksize that could handle larger arrays, but wouldn't because its stacksize isn't set to unlimited.